var/spack/repos/builtin/packages/openradioss/package.py

```diff
@@ -0,0 +1,176 @@
+
+        if debug == "2":
+            sanitize = "1"
+        else:
+            sanitize = "0"
+
+        if "+mpi" in spec:
+            mpi_def = [
+                self.define("MPI", "ompi"),
+                self.define("mpi_root", spec["mpi"].prefix),
+                self.define("mpi_libdir", spec["mpi"].prefix.lib),
+                self.define("mpi_incdir", spec["mpi"].prefix.include),
+            ]
+        else:
+            mpi_def = [self.define("MPI", "smp")]
+
+        args = [
+            self.define_from_variant("arch", "arch"),
+            self.define_from_variant("precision", "precision"),
+            self.define_from_variant("debug", "debug"),
+            self.define_from_variant("static_link", "static-link"),
+            self.define("sanitize", sanitize),
+        ] + mpi_def
+
+        return args
+
+    # Openradioss contains two separate cmake projects
+    # We install "engine" using the default cmake(), build() provided by
+    # CmakePackage and "starter" by repeating those two functions.
+    @property
+    def root_cmakelists_dir(self):
+        return "engine"
+
+    @property
+    def build_directory(self):
+        return os.path.join(self.stage.source_path, self.root_cmakelists_dir, "build_dir")
+
+    # Run cmake in "starter" directory as well.
+    # This is a copy of lib/spack/spack/build_systems.py:cmake()
+    @run_before("cmake")
+    def cmake_starter(self):
+        """Runs ``cmake`` in the build directory"""
+        options = self.std_cmake_args
+        options += self.cmake_args()
+        options.append(os.path.abspath("starter"))
+        with working_dir(
+            os.path.join(self.stage.source_path, "starter", "build_dir"), create=True
+        ):
+            inspect.getmodule(self).cmake(*options)
+
+    # Run build in "starter" directory as well.
+    # This is a copy of lib/spack/spack/build_systems.py:build()
+    @run_before("build")
+    def build_starter(self):
+        """Make the build targets"""
+        with working_dir(os.path.join(self.stage.source_path, "starter", "build_dir")):
+            if self.generator == "Unix Makefiles":
+                inspect.getmodule(self).make(*self.build_targets)
+            elif self.generator == "Ninja":
+                self.build_targets.append("-v")
+                inspect.getmodule(self).ninja(*self.build_targets)
+
+    def install(self, spec, prefix):
+        """Installation is just moving executables."""
+        if spec.variants["debug"].value == "1":
+            ddebug = "_db"
+        else:
+            ddebug = ""
+        if spec.variants["precision"].value == "sp":
+            suffix = "_sp"
+        else:
+            suffix = ""
+        extension = "_" + spec.variants["arch"].value + suffix + ddebug
+        os.mkdir(self.prefix.bin)
+        os.mkdir(self.prefix.lib64)
+
+        move(
+            os.path.join(self.stage.source_path, "starter", "build_dir", "starter" + extension),
```

Comment:
FYI. Spack has a `join_path()` routine (and a `mkdirp()`) already imported.

Reply:
Thanks, Done.